### DISCUSSION TITLE
Trying to install  linux-surface on MS Surface 3

### DISCUSSION THREAD

**Discussion by writersglen** [linux-surface/linux-surface#488] [Q&A]
In attempt to import keys to sign surface-latex packages I get this error message:

gpg: no valid OpenPGP data found.

Checked—gnupg is installed.  (2.2.19-3ubuntu2.1)

Googled gpg no valid openpgp data found + surface

Problem shows up in many posts, but no solutions that worked for me.

I’m missing something. Can someone please guide me through.

Many thanks,

LRP

**Comment by qzed** [comment]
What commands did you run specifically? I assume
```
wget -qO - https://raw.githubusercontent.com/linux-surface/linux-surface/master/pkg/keys/surface.asc \
    | gpg --dearmor | sudo dd of=/etc/apt/trusted.gpg.d/linux-surface.gpg
```

Can you try splitting them up into
```
wget https://raw.githubusercontent.com/linux-surface/linux-surface/master/pkg/keys/surface.asc
```
and
```
cat surface.asc | gpg --dearmor | sudo dd of=/etc/apt/trusted.gpg.d/linux-surface.gpg
```

to make sure that the key is downloaded properly (i.e. check that `surface.asc` exists and is not empty)?

**Comment by writersglen** [comment]
Many thanks Maximilan,

I’ll try this when I get back on the project later this week.

Since the hurdle seems to be related to PGP, I’ve been wondering if I need a public PGP signing key. But that that seems like circular logic—need a key to get a key?

All the best,

Lloyd

**Reply by qzed** [reply]
You only need to create a key yourself if you want to sign something. For installing, you don't need to create one, you just need to import our public key so that the package manager can validate the packages.

**Comment by writersglen** [comment]
Thanks Maximilian. You’re the best. And a big thanks to all who’ve put so much work into bringing Linux to Surface devices.

It’ll be Wednesday before I can get back to the task, but hope springs eternal.

All the best,

Lloyd

**Reply by writersglen** [reply]
just successfully installed Ubuntu 20.04 on my MS Surface 3.

Many thanks to Maxmillian and all who’ve made this possible!

Now to test. Will it live up to my hopes?

All the best,

LRP
